### PR TITLE
Solid biomass to hydrogen process

### PR DESCRIPTION
**Pull request by millingermarkus**
Closes # (if applicable).

## Changes proposed in this Pull Request
Added the solid biomass to hydrogen process, which converts solid biomass to hydrogen while capturing the biogenic carbon.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Review by lindnemi — APPROVED** [review]
Needed  for Ariadne

File: doc/release_notes.rst [hunk 1]
```diff
@@ -9,6 +9,9 @@ Release Notes
 
 .. Upcoming Release
 .. ================
+
+* Add option to produce hydrogen from solid biomass (flag ``solid biomass to hydrogen`), combined with carbon capture
```
Comment (fneum) [inline doc/release_notes.rst:13]:
```suggestion
* Add option to produce hydrogen from solid biomass (flag ``solid biomass to hydrogen``), combined with carbon capture
```

File: scripts/prepare_sector_network.py [hunk 2]
```diff
@@ -2543,6 +2543,27 @@ def add_biomass(n, costs):
...
+            bus0=spatial.biomass.nodes,
+            bus1=spatial.h2.nodes,
+            bus2=spatial.co2.nodes,
+            bus3="co2 atmosphere",
+            carrier="solid biomass to hydrogen",
+            efficiency=costs.at['solid biomass to hydrogen', 'efficiency'],
+            efficiency2=costs.at['solid biomass', 'CO2 intensity'] * options["cc_fraction"],
+            efficiency3=-costs.at['solid biomass', 'CO2 intensity'] * options["cc_fraction"],
+            p_nom_extendable=True,
+            capital_cost=costs.at['solid biomass to hydrogen', 'fixed'] * costs.at['solid biomass to hydrogen', 'efficiency']
```
Comment (fneum) [inline scripts/prepare_sector_network.py:2562]: Just to double-check, the unit in https://github.com/PyPSA/technology-data/blob/523eea9f36867da8fcf5e1e2aa91f1d4fe9a667b/outputs/costs_2030.csv#L1026 is €/kW_th H2?

**Review by fneum — APPROVED** [review]
LGTM, just one double-check of the capital cost unit required.

File: scripts/prepare_sector_network.py (hunk 2 above)
Comment (lindnemi) [reply scripts/prepare_sector_network.py:2562]: just remembered that we now always need to report the investments as well, i.e. 
```
overnight_cost=costs.at['solid biomass to hydrogen', 'investment'] * costs.at['solid biomass to hydrogen', 'efficiency'] + costs.at['biomass CHP capture', 'investment'] * costs.at['solid biomass', 'CO2 intensity'],

```

**Review by lindnemi — CHANGES_REQUESTED** [review]
just remembered that we now always need to report the investments as well, i.e. 
```
overnight_cost=costs.at['solid biomass to hydrogen', 'investment'] * costs.at['solid biomass to hydrogen', 'efficiency'] + costs.at['biomass CHP capture', 'investment'] * costs.at['solid biomass', 'CO2 intensity'],

```

**Comment by fneum** [comment]
That was only merged into `ariadne2` so far, wasn't it?

File: scripts/prepare_sector_network.py (hunk 2 above)
Comment (millingermarkus) [reply scripts/prepare_sector_network.py:2562]: > Just to double-check, the unit in https://github.com/PyPSA/technology-data/blob/523eea9f36867da8fcf5e1e2aa91f1d4fe9a667b/outputs/costs_2030.csv#L1026 is €/kW_th H2?

Yes:
image
Source: Zech et.al. 2014: https://www.dbfz.de/fileadmin/user_upload/Referenzen/DBFZ_Reports/DBFZ_Report_19.pdf

**Review by fneum — APPROVED** [review]
Great, cheers!

**Pull request merged**